src/pagination/components/Pagination.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+class Pagination extends React.Component{
+    constructor(props){
+        super(props);
+        this.state={
+            allDataRowsNumber: 0,
+            currentPage: 1,
+            dataPerPage: props.dataPerPage
+        }
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({ allDataRowsNumber: nextProps.allDataRowsNumber });  
```

Comment:
From the [official documentation](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops):

> (...) invoked before a mounted component receives new props. If you need to update the state in response to prop changes (for example, to reset it), you may compare `this.props` and `nextProps` and perform state transitions using `this.setState()` in this method.

This is not the case here, since the state transition doesn't depend on current props. You can simply pass `allDataRowsNumber` as a prop from the parent component.

Also, this method has been deprecated, and is called `UNSAFE_componentWilReceiveProps` in the latest version of React.

Reply:
If I remove it pagination receives allDataRowsNumber 0 on the first render and does not appear.

Reply:
I made it work now with the shouldComponentUpdate method. Is that OK?

Reply:
Dunno; I'll have to see. ¯\_(ツ)_/¯